### PR TITLE
Remove the obsolete Repeated API

### LINKED ISSUE #1296: Remove the Repeated API in 6.0.0
Related to #1292, #808, #1295

### PR DESCRIPTION
**Pull request by thomaslevesque**
Fixes #1296

### CODE REVIEW

**Comment by thomaslevesque** [comment]
I made 2 separate commits, in case we want to keep the FakeItEasy006 diagnostic.

File: tests/FakeItEasy.Tests/Configuration/AnyCallConfigurationTests.cs [hunk 1]
```diff
@@ -204,19 +204,17 @@ public void AssignsOutAndRefParametersLazily_returns_configuration_produced_by_f
...
             var factoryConfig = this.StubVoidConfig();
 
             // Act
-            this.configuration.MustHaveHappened(Repeated.AtLeast.Twice);
+            this.configuration.MustHaveHappenedTwiceOrMore();
 
             // Assert
-            A.CallTo(() => factoryConfig.MustHaveHappened(A<Repeated>._)).MustHaveHappened();
+            A.CallTo(() => factoryConfig.MustHaveHappened(2, Times.OrMore)).MustHaveHappened();
         }
```
[the remark this answers is not in the document]
Comment (thomaslevesque) [reply tests/FakeItEasy.Tests/Configuration/AnyCallConfigurationTests.cs]: I don't mind deleting it. I don't even understand what it tests...
At some point, we should probably do a big clean up of the unit tests anyway. Most of them are [Gulliver's travels tests](http://dhickey.ie/2014/03/03/gullivers-travels-test/).

[the remark this answers is not in the document]
Comment (blairconrad) [reply tests/FakeItEasy.Tests/Configuration/AnyCallConfigurationTests.cs]: It was a good day when M. Ralph introduced that image to us. Or at least me. You might've been familiar already.

**Comment by thomaslevesque** [comment]
Thanks for the merge!

**Pull request merged**